pkg/kubectl/cmd/util/printing.go

```diff
@@ -81,7 +83,10 @@ func OutputVersion(cmd *cobra.Command, defaultVersion string) string {
 // Requires that printer flags have been added to cmd (see AddPrinterFlags).
 func PrinterForCommand(cmd *cobra.Command) (kubectl.ResourcePrinter, bool, error) {
 	outputFormat := GetFlagString(cmd, "output")
-	templateFile := GetFlagString(cmd, "template")
+	templateFile := GetFlagString(cmd, "golang-template")
```

Comment:
Can shell completion deal with `-o <prefix>`?

`go-template` would be ok, rather than `golang-template`, and `jsonpath` rather than `jsonpath-template`. 

"jp" and "tmpl" are a bit too obscure, except maybe as aliases for the verbose forms.

cc @daizouzhou, since this thread is otherwise hard to find